salt/utils/versions.py

```diff
@@ -395,24 +375,32 @@ def check_boto_reqs(
     various, and many, boto modules and states.
 
     boto_ver
-        The minimum required version of the boto library. Defaults to ``2.0.0``.
+        The minimum required version of the boto library. Conflicts with boto3_ver.
 
     boto3_ver
-        The minimum required version of the boto3 library. Defaults to ``1.2.6``.
+        The minimum required version of the boto3 library. Conflicts with boto_ver.
 
     botocore_ver
         The minimum required version of the botocore library. Defaults to ``1.3.23``.
 
-    check_boto
-        Boolean defining whether or not to check for boto deps. This defaults to ``True`` as
-        most boto modules/states rely on boto, but some do not.
-
-    check_boto3
-        Boolean defining whether or not to check for boto3 (and therefore botocore) deps.
-        This defaults to ``True`` as most boto modules/states rely on boto3/botocore, but
-        some do not.
     """
-    if check_boto is True:
+
+    if check_boto is not None:
+        salt.utils.versions.warn_until(
+            "Aluminium",
+            "The 'check_boto' argument has been deprecated and its functionality removed, as such, its usage is no longer required.",
```

Comment:
We can't remove its functionality yet until the release we remove it from, so you will need to ensure the functionality of those kwargs still works to ensure backwards compatibility until removal.

Also the release name we would want to warn until is `Phosphorus` since this will be merged in for Aluminium.

Reply:
Hmm .. I need to figure out a way to rethink this PR then Generally speaking, no one should ever have used boto and boto3 at the same time. But, I will see about refactoring this PR to continue allowing it.

Reply:
New changes should allow the old pattern of allowing `check_boto` and `check_boto3` while also supporting the new pattern of specifying a version to enable checking for that version.